### ISSUE TITLE
[BUG] apps won't load due to missing module "typing_extension"

### ISSUE THREAD

**Issue by Crocmagnon** [xaviml/controllerx#191]
# Bug report

## Description

ControllerX apps don't load because of a missing module (`typing_extensions`, see the logs below).

## Additional information

* Devices involved:
  * Model: Legrand 600083 as Light controller
* Integration: `zha`
* AppDaemon version: 4.0.3
* ControllerX version: 4.2.0
* HACS version (if installed from there): 1.6.2
* Home Assistant Core version: 0.118.4

## AppDaemon app configuration

```yaml
bureau:
  module: controllerx
  class: Legrand600083LightController
  controller: 00:04:74:00:00:ad:38:38
  integration: zha
  light: light.bureau
```

## Logs

```text
Traceback (most recent call last):
Dec 05 15:51:56 architect appdaemon[31050]:   File "/srv/appdaemon/lib/python3.7/site-packages/appdaemon/app_management.py", line 820, in check_app_updates
Dec 05 15:51:56 architect appdaemon[31050]:     await self.init_object(app)
Dec 05 15:51:56 architect appdaemon[31050]:   File "/srv/appdaemon/lib/python3.7/site-packages/appdaemon/app_management.py", line 271, in init_object
Dec 05 15:51:56 architect appdaemon[31050]:     modname = await utils.run_in_executor(self, __import__, app_args["module"])
Dec 05 15:51:56 architect appdaemon[31050]:   File "/srv/appdaemon/lib/python3.7/site-packages/appdaemon/utils.py", line 276, in run_in_executor
Dec 05 15:51:56 architect appdaemon[31050]:     response = future.result()
Dec 05 15:51:56 architect appdaemon[31050]:   File "/usr/lib/python3.7/concurrent/futures/thread.py", line 57, in run
Dec 05 15:51:56 architect appdaemon[31050]:     result = self.fn(*self.args, **self.kwargs)
Dec 05 15:51:56 architect appdaemon[31050]:   File "/home/homeassistant/.homeassistant/appdaemon/apps/controllerx/controllerx.py", line 6, in <module>
Dec 05 15:51:56 architect appdaemon[31050]:     from cx_core import (
Dec 05 15:51:56 architect appdaemon[31050]:   File "/home/homeassistant/.homeassistant/appdaemon/apps/controllerx/cx_core/__init__.py", line 2, in <module>
Dec 05 15:51:56 architect appdaemon[31050]:     from cx_core.custom_controller import (
Dec 05 15:51:56 architect appdaemon[31050]:   File "/home/homeassistant/.homeassistant/appdaemon/apps/controllerx/cx_core/custom_controller.py", line 3, in <module>
Dec 05 15:51:56 architect appdaemon[31050]:     from cx_core.type.light_controller import LightController
Dec 05 15:51:56 architect appdaemon[31050]:   File "/home/homeassistant/.homeassistant/appdaemon/apps/controllerx/cx_core/type/light_controller.py", line 14, in <module>
Dec 05 15:51:56 architect appdaemon[31050]:     from typing_extensions import Literal
Dec 05 15:51:56 architect appdaemon[31050]: ModuleNotFoundError: No module named 'typing_extensions'
```

## Additional Context

Add any other context or screenshots about the bug here.

**Comment by Crocmagnon** [comment]
Installing `typing_extensions` in the virtualenv where appdaemon is installed works around this issue.

**Comment by xaviml** [comment]
Hi @Crocmagnon,

Thank you for reporting this. In fact, I changed this in the dev branch, I will deploy the fix later. Did you install AppDaemon with docker? Which Python version are you using?

Regards,
Xavi M.

**Comment by xaviml** [comment]
Hi @Crocmagnon,

This has been fixed in ControllerX [v4.2.1](https://github.com/xaviml/controllerx/releases/tag/v4.2.1).

Regards,
Xavi M.

**Comment by Crocmagnon** [comment]
> Did you install AppDaemon with docker?

No

> Which Python version are you using?

3.7.3

> This has been fixed in ControllerX v4.2.1.

Thanks for your quick fix!
